docs/network-troubleshooting.adoc

```diff
@@ -0,0 +1,143 @@
+:toc: macro
+
+= Network troubleshooting
+
+:icons: font
+:numbered:
+toc::[]
+
+Clients forming the Keep Network are deployed in numerous ways using different
+infrastructure configurations and network topologies. This fact combined with
+the complexity of a p2p network may cause `keep-ecdsa` clients to experience
+some network problems occasionally. Those issues are typically noticed during
+key generation or signing processes which doesn't succeed at the first time
+and are retried. Usually, the problem is caused by clients going offline
+temporarily due to restart or network breaks. In that case, clients should
+recover automatically as `keep-ecdsa` clients are resilient and can
+continue their work after reconnecting to the network or coming up after the
+restart. However, sometimes the problem persists and troubleshooting process
+should be run on all clients taking part in the unlucky protocol execution.
+
+== Common problems
+
+Typically, there are several types of network-related problems reported by
+operators. Here is a brief description of them.
+
+=== Failed to get unicast channel with peer
+
+During key generation and signing, all keep members need to open a direct
+unicast channel between each other. If, from whatever reason, opening such
+a channel is not possible, the client will print a warning looking like:
+
+```
+failed to get unicast channel with peer [peer_id] because of: [cause]
+```
+
+=== Failed to announce signer presence
+
+Announce protocol is performed before the key generation process. It's purpose
+is to exchange public keys between members. Public keys are then used to
+derive the peers transport identifiers necessary to open unicast channels.
+Announce protocol duration is capped with a 2 minutes timeout. In case the
+timeout passes and the client doesn't receive public keys from all of other
+keep members, the following error will be printed in logs:
+
+```
+failed to announce signer presence: [waiting for announcements timed out after: [2m0s]]"
+```
+
+=== Readiness signaling protocol failed
+
+Readiness protocol is performed before both key generation and signing
+processes. It's purpose is to synchronize keep members and allows each member
+to signal their readiness to perform the TSS protocol. Like the announce
```

Comment:
I would replace "TSS" with just "key generation/signing". I do not think anyone but core developers know what TSS stands for.

Reply:
Good catch, see https://github.com/keep-network/keep-ecdsa/pull/838/commits/c37ea1b3298859422a185d36b928ede271cbd4fe